cvmfs/sync_mediator.cc

```diff
@@ -226,6 +226,8 @@ manifest::Manifest *SyncMediator::Commit() {
     return NULL;
   }
 
+  if (catalog_manager_->IsBalanced())
```

Comment:
This is correct... even though maybe the method is not descriptive enough.
What I mean is that if the WritableCatalogManager IS balanceable, then proceed with the balancing operation.
Maybe the method should be renamed to CanBeBalanced()...